app/src/main/java/com/nutomic/syncthingandroid/fragments/DrawerFragment.java

```diff
@@ -211,15 +214,19 @@ private void showQrCode() {
             Toast.makeText(mActivity, R.string.syncthing_terminated, Toast.LENGTH_SHORT).show();
             return;
         }
-        String apiKey = restApi.getGui().apiKey;
-        String deviceId = restApi.getLocalDevice().deviceID;
-        URL url = restApi.getUrl();
-        //The QRCode request takes one paramteer called "text", which is the text to be converted to a QRCode.
-        new ImageGetRequest(mActivity, url, ImageGetRequest.QR_CODE_GENERATOR, apiKey,
-                ImmutableMap.of("text", deviceId),qrCodeBitmap -> {
-            mActivity.showQrCodeDialog(deviceId, qrCodeBitmap);
-            mActivity.closeDrawer();
-        }, error -> Toast.makeText(mActivity, R.string.could_not_access_deviceid, Toast.LENGTH_SHORT).show());
+        try {
+            String apiKey = restApi.getGui().apiKey;
+            String deviceId = restApi.getLocalDevice().deviceID;
+            URL url = restApi.getUrl();
+            //The QRCode request takes one paramteer called "text", which is the text to be converted to a QRCode.
+            new ImageGetRequest(mActivity, url, ImageGetRequest.QR_CODE_GENERATOR, apiKey,
+                    ImmutableMap.of("text", deviceId),qrCodeBitmap -> {
+                mActivity.showQrCodeDialog(deviceId, qrCodeBitmap);
+                mActivity.closeDrawer();
+            }, error -> Toast.makeText(mActivity, R.string.could_not_access_deviceid, Toast.LENGTH_SHORT).show());
+        } catch (Exception e) {
+            Log.e(TAG, "showQrCode", e);
```

Comment:
Do we need to show something to the user, or will it be obvious once the drawer is closed?

Reply:
I think in most cases a problem occurs here we'll hit the part that shows the toast to the user. That's enough for me.
> if (restApi == null) {
 -        String deviceId = mActivity.getApi().getLocalDevice().deviceID;		 +            Toast.makeText(mActivity, R.string.syncthing_terminated, Toast.LENGTH_SHORT).show();
 -        URL url = mActivity.getApi().getUrl();		 +            return;
 -        new ImageGetRequest(mActivity, url, ImageGetRequest.QR_CODE_GENERATOR, apiKey,		 +        }